ibm/service/cis/resource_ibm_cis_origin_auth.go

```diff
@@ -0,0 +1,322 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+package cis
+
+import (
+	"context"
+	"fmt"
+
+	"log"
+	"strings"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/networking-go-sdk/authenticatedoriginpullapiv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	cisOriginAuthID          = "auth_id"
+	cisOriginAuthHost        = "hostname"
+	cisOriginAuthEnable      = "enabled"
+	cisOriginAuthLevel       = "level"
+	cisOriginAuthCertContent = "certificate"
+	cisOriginAuthCertKey     = "private_key"
+	cisOriginAuthCertId      = "cert_id"
+	CisOriginAuthStatus      = "status"
+	cisOriginAuthExpiresOn   = "expires_on"
+	cisOriginAuthUploadedOn  = "uploaded_on"
+)
+
+func ResourceIBMCISOriginAuthPull() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMCISOriginAuthPullCreate,
+		ReadContext:   resourceIBMCISOriginAuthPullRead,
+		UpdateContext: resourceIBMCISOriginAuthPullUpdate,
+		DeleteContext: resourceIBMCISOriginAuthPullDelete,
+		Importer:      &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisOriginAuthLevel: {
+				Type:        schema.TypeString,
+				Description: "Origin auth level zone or hostname",
+				Required:    true,
+			},
+			cisOriginAuthHost: {
+				Type:        schema.TypeString,
+				Description: "Host name needed for host level authentication",
+				Optional:    true,
+			},
+			cisOriginAuthEnable: {
+				Type:        schema.TypeBool,
+				Description: "Enabel-disable origin auth for a zone or host",
+				Optional:    true,
+				Default:     true,
+			},
+			cisOriginAuthCertContent: {
+				Type:        schema.TypeString,
+				Description: "Certificate content which needs to be uploaded",
+				Required:    true,
+			},
+			cisOriginAuthCertKey: {
```

Comment:
Can you mark those attributes which are sensitive and which should not be displayed on console...by adding 
Sensitive:true
https://www.terraform.io/plugin/sdkv2/best-practices/sensitive-state

Reply:
oh, I was not aware of this flag.. thanks for pointing this out.